src/test/kotlin/org/rust/lang/core/resolve/RsResolveTest.kt

```diff
@@ -1193,4 +1193,24 @@ class RsResolveTest : RsResolveTestBase() {
             }     //^
         }
     """)
+
+    fun `test extern crate self`() = checkByCode("""
+        extern crate self as foo;
+
+        struct Foo;
+              //X
+
+        use foo::Foo;
+                //^
+    """)
+
+    fun `test extern crate self without alias`() = checkByCode("""
+        extern crate self;
```

Comment:
Looks like this test don't test the feature b/c it should pass without `extern crate self;`

Reply:
Oh, or it tests that behavior of `self` isn't changed by adding (wrong) `extern crate self;`

Reply:
> Oh, or it tests that behavior of self isn't changed by adding (wrong) extern crate self;

Yep